augly/video/functional.py

```diff
@@ -138,6 +141,65 @@ def audio_swap(
     return output_path or video_path
 
 
+def augment_audio(
+    video_path: str,
+    audio_aug: Callable[..., Tuple[np.ndarray, int]] = audaugs.apply_lambda,
+    output_path: Optional[str] = None,
+    metadata: Optional[List[Dict[str, Any]]] = None,
+    **audio_aug_kwargs,
+) -> str:
+    """
+    Augments the audio track of the input video using a given AugLy audio augmentation
+
+    @param video_path: the path to the video to be augmented
+
+    @param audio_aug: the augmentation function to be applied onto the video's audio
+        track. Should have the standard API of an AugLy audio augmentation, i.e. expect input audio
+        as a numpy array or path & output path as input, and output the augmented audio to the
+        output path
+
+    @param output_path: the path in which the resulting video will be stored.
+        If not passed in, the original video file will be overwritten
+
+    @param metadata: if set to be a list, metadata about the function execution
+        including its name, the source & dest duration, fps, etc. will be appended
+        to the inputted list. If set to None, no metadata will be appended or returned
+
+    @param audio_aug_kwargs: the input attributes to be passed into `audio_aug`
+
+    @returns: the path to the augmented video
+    """
+    assert callable(audio_aug), (
+        repr(type(audio_aug).__name__) + " object is not callable"
+    )
+
+    func_kwargs = helpers.get_func_kwargs(metadata, locals(), video_path)
+
+    audio_metadata = []
+
+    with tempfile.NamedTemporaryFile(suffix=".aac") as tmpfile:
+
+        helpers.extract_audio_to_file(video_path, tmpfile.name)
+        aug_audio, aug_sr = audio_aug(tmpfile.name)
+        audutils.ret_and_save_audio(aug_audio, tmpfile.name, aug_sr)
+        audio_swap(video_path, tmpfile.name)
+
+        if metadata is not None:
+            helpers.get_metadata(
+                metadata=audio_metadata, function_name="augment_audio", **func_kwargs
+            )
```

Comment:
```suggestion
```

We can actually get the audio metadata by just passing in the `metadata` arg to the audio augmentation function (see my suggested change on line 183). This wouldn't work anyway because you're calling the video `get_metadata()` function, which is different from the audio `get_metadata()` function which will be called inside the audio augmentation function.